src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs

```diff
@@ -0,0 +1,164 @@
+        [Alias("CN", "__SERVER", "Server", "IPAddress")]
+        public string[] ComputerName { get; set; } = new string[] { "." };
+
+        /// <summary>
+        /// The following is the definition of the input parameter "Credential".
+        /// Specifies a user account that has permission to perform this action. Type a
+        /// user-name, such as "User01" or "Domain01\User01", or enter a PSCredential
+        /// object, such as one from the Get-Credential cmdlet.
+        /// </summary>
+        [Parameter(Position = 1)]
+        [ValidateNotNullOrEmpty]
+        [Credential]
+        public PSCredential Credential { get; set; }
+
+        /// <summary>
+        /// Force the operation to take place if possible.
+        /// </summary>
+        [Parameter]
+        public SwitchParameter Force { get; set; } = false;
+
+        #endregion "parameters"
+
+        #region "IDisposable Members"
+
+        /// <summary>
+        /// Dispose Method.
+        /// </summary>
+        public void Dispose()
+        {
+            try
+            {
+                _cancel.Dispose();
+            }
+            catch (ObjectDisposedException) { }
+        }
+
+        #endregion "IDisposable Members"
+
+        #region "Overrides"
+
+        /// <summary>
+        /// ProcessRecord.
+        /// </summary>
+        [SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
+        protected override void ProcessRecord()
+        {
+            if (ComputerName.Length > 1) {
+                ErrorRecord error = new ErrorRecord(
+                        new InvalidOperationException("Only localhost is currently supported"), "InvalidNewName",
+                        ErrorCategory.InvalidArgument, ComputerName);
+                WriteError(error);
+            }
+            if (ComputerName[0].Equals(".") || ComputerName[0].Equals("localhost")) {
+                doShutdown();
+                return;
+            }
+            WriteError(new ErrorRecord(
+                        new InvalidOperationException("Only localhost is currently supported"), "InvalidNewName",
+                        ErrorCategory.InvalidArgument, ComputerName));
+        }
+
+        /// <summary>
+        /// To implement ^C.
+        /// </summary>
+        protected override void StopProcessing()
+        {
+            try
+            {
+                _cancel.Cancel();
+            }
+            catch (ObjectDisposedException) { }
+            catch (AggregateException) { }
+        }
+
+        #endregion "Overrides"
+
+        #region "Internals"
+        #if OSX
+        [DllImport("libSystem.dylib")]
+        private static extern int system(string cmd);
+        #else
+        [DllImport("libc.so.6")]
+        private static extern int system(string cmd);
+        #endif
+
+        private void doShutdown() {
+            int result = system("shutdown -P now");
```

Comment:
How do we handle it when the user doesn't have permissions to shutdown?

Reply:
The command will error out and we (now) write the error code to the stream.